src/genbenchQC/utils/data_leakage_utils.py

```
@@ -0,0 +1,173 @@
+
+        expected_q = qseq[qstart0:qend]
+        expected_t = tseq[tstart0:tend]
+
+        ungapped_qaln = qaln.replace("-", "")
+        ungapped_taln = taln.replace("-", "")
+
+        if ungapped_qaln != expected_q:
+            raise ValueError(
+                f"Query alignment mismatch:\n"
+                f"  Ungapped aligned sequence does not match original sequence slice."
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)